components/dashboard/src/data/workspaces/fetchers.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { GitpodServer, WorkspaceInfo } from "@gitpod/gitpod-protocol";
+import { useCallback } from "react";
+import { useFeatureFlags } from "../../contexts/FeatureFlagContext";
+import { workspacesService } from "../../service/public-api";
+import { getGitpodService } from "../../service/service";
+
+type UseWorkspacesFetcherArgs = {
+    limit: number;
+};
+
+export type WorkspacesFetcherResult = WorkspaceInfo[];
+
+export const useWorkspacesFetcher = ({ limit = 50 }: UseWorkspacesFetcherArgs) => {
```

Comment:
The more I write the word `fetcher` the less I like it 🤣 

As the intent of the separation between queries, and these fetchers, is to abstract the react query hooks from ones that call our apis, maybe a name like API would be better?

Instead of:

* `fetchers.ts` => `api.ts`
* `useWorkspacesFetcher` => `useWorkspacesAPI`
* ...etc

Thoughts?

Reply:
I would find it more helpful if we were naming the queries and mutation files after what they do (e.g. `delete-workspace-mutation.ts`),  but then also include the fetching logic there. WDYT?

Reply:
Yah that's a good callout on the separation of query/fetcher. My motivation for separating that logic was that it couples us slightly less to `react-query` - could pretty easily switch to something like `swr` and the fetcher function wouldn't have to change. That theoretical benefit is probably not worth it if it over-complicates things (and I tend to agree that it does, makes setting things up feel a lot heavier).

I'll take a stab at removing the fetcher piece and doing the api calls inline in the query/mutation functions and see how it feels.

Regarding splitting query/mutations into more discrete files - sure I'm not opposed to that, I wasn't sure how often we'd have multiple of each, so opted for a single file for each type. I can give that a shot too and see how it feels with the above change.